include/deal.II/lac/read_write_vector.h

```diff
@@ -517,6 +535,35 @@ namespace LinearAlgebra
      * Make all other ReadWriteVector types friends.
      */
     template <typename Number2> friend class ReadWriteVector;
+
+#ifdef DEAL_II_WITH_CXX11
+  private:
+    template <typename Functor>
+    class FunctorTemplate
+    {
+    public:
+      /**
+       * Constructor. Take a functor and store a copy of it.
+       */
+      FunctorTemplate(ReadWriteVector<Number> &parent, const Functor &functor);
+
+      /**
+       * Evaluate the element with the stored copy of the functor.
+       */
+      virtual void operator() (const size_type begin, const size_type end);
+
+    private:
+      /**
+       * Alias to the ReadWriteVector object that owns the FunctorTemplate.
+       */
+      ReadWriteVector &parent;
+
+      /**
+       * Copy of the functor.
+       */
+      Functor functor;
```

Comment:
I imagine that you could store a const reference here, since the lifetime of the enclosing object is likely shorter than that of the `functor` object, no?